### PR TITLE
[DSIP-39][parameter] Improvement startup parameters/global parameters/project parameters data type

### LINKED ISSUE #15936: [DSIP-39][parameter] Startup parameters/global parameters/project parameters support set data type
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Motivation
Currently startup parameters, global parameters, project parameters do not support custom data value types, the default is VARCHAR data value type, if I want to use the data type is LONG, this time there is no way to carry out the operation, so this optimization point is to give more choice points to the user, to increase the user experience.

### Design Detail
一、 DESC：
This PR is for the startup parameter, global parameter, project parameter support custom data value type, to meet our needs for data use, increase user experience. Which I did not modify the logic of the big, just load the data value type to personalize the limited operation, specifically the following points:
- 1.1 the default data type of startup parameters, global parameters, project parameters is VARCHAR, then this time I transformed this, support for custom data types and compatible with the original data type operation.
- 1.2 StartParams was originally MAP way, now changed to LIST, which I judged whether it is JSON compatible operation.
- 1.3 add New UI
- 1.4 add UT

二、 EFFECT DISPLAY：
2.1 GIF

2.2 Startup Parameters
启动参数
2.3 Global Parameters
全局参数
2.4 Project Parameters
项目参数

### Compatibility, Deprecation, and Migration Plan

Compatibility with latest version.

### Test Plan

Test by UT.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by sdhzwc**
close #15936 
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by SbloodyS — COMMENTED** [review]
Please add the docs. @sdhzwc

**Comment by sdhzwc** [comment]
> Please add the docs。 @sdhzwc

Yesterday I got feedback that the front-end compilation failed, so I'll improve the documentation when I've solved this problem.

**Comment by SbloodyS** [comment]
> Yesterday I got feedback that the front-end compilation failed, so I'll improve the documentation when I've solved this problem.

`pnpm run build:prod` command failed. You can test it locally.

**Comment by ruanwenjun** [comment]
I tried to compile the frontend locally but failed.

image

**Comment by sdhzwc** [comment]
> I tried to compile the frontend locally but failed.
> 
> image



> I tried to compile the frontend locally but failed.
> 
> image

My local is windows environment, using the same method, but reported a bunch of errors, can not lock the problem, but by watching you send the error, I found that onClearSearchTaskType method is not used, I removed, it should be good, or you run a little!

**Comment by sdhzwc** [comment]
> > Yesterday I got feedback that the front-end compilation failed, so I'll improve the documentation when I've solved this problem.
> 
> `pnpm run build:prod` command failed. You can test it locally.

done

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/expand/CuringParamsServiceImpl.java [hunk 1]
```diff
@@ -150,9 +154,17 @@ public Map<String, Property> parseWorkflowStartParam(@Nullable Map<String, Strin
...
+        JsonElement jsonElement = JsonParser.parseString(startParamJson);
+        // check whether it is json
+        boolean isJson = jsonElement.isJsonObject();
+        if (isJson) {
+            Map<String, String> startParamMap = JSONUtils.toMap(startParamJson);
+            return startParamMap.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey,
+                    entry -> new Property(entry.getKey(), Direct.IN, DataType.VARCHAR, entry.getValue())));
+        } else {
+            List<Property> propList = JSONUtils.toList(startParamJson, Property.class);
+            return propList.stream().collect(Collectors.toMap(Property::getProp, Function.identity()));
```
Comment (ruanwenjun) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/expand/CuringParamsServiceImpl.java]: Use DataTransformUtils?

**Review by ruanwenjun — COMMENTED** [review]
The database schema sync check failed with pg

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/expand/CuringParamsServiceImpl.java (hunk 1 above)
Comment (sdhzwc) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/expand/CuringParamsServiceImpl.java]: > Use DataTransformUtils?

One is an api, the other is a service. 
Do you mean I need to write another one in the service?

**Comment by sdhzwc** [comment]
> The database schema sync check failed with pg

image
I don't see an error related to the code I wrote
I successfully added the SQL statement.

**Comment by ruanwenjun** [comment]
> > The database schema sync check failed with pg
> 
> image I don't see an error related to the code I wrote I successfully added the SQL statement.

This is a history problem, not related to this PR.

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/expand/CuringParamsServiceImpl.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/expand/CuringParamsServiceImpl.java]: Can we use DataTransformUtils to transform the `startParamJson ` to List<Property>?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/DataTransformUtils.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+
+/**
+ * data transform utils
+ */
+public class DataTransformUtils {
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/DataTransformUtils.java]:
```suggestion
public class PropertyUtils {
```

Comment (ruanwenjun) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/DataTransformUtils.java]: It's better to use a specific name.

**Comment by ruanwenjun** [comment]
> > The database schema sync check failed with pg
> 
> image I don't see an error related to the code I wrote I successfully added the SQL statement.

This may due to a history problem, I submit https://github.com/apache/dolphinscheduler/pull/15995 to fix this.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/DataTransformUtils.java (hunk 2 above)
Comment (sdhzwc) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/DataTransformUtils.java]: > It's better to use a specific name.

done

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/expand/CuringParamsServiceImpl.java (hunk 1 above)
Comment (sdhzwc) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/expand/CuringParamsServiceImpl.java]: > Can we use DataTransformUtils to transform the `startParamJson ` to List?

done

**Review by ruanwenjun — DISMISSED** [review]
LGTM, wait the ci pass

**Comment by ruanwenjun** [comment]
Since the next release will from dev, so I change the milestone to 3.2.2, otherwise this pr cannot be merged before we release 3.2.2

**Review by songjianet — DISMISSED** [review]

**Comment by sdhzwc** [comment]
@ruanwenjun I see someone merged in before me, but I can't pull the code from my side, is this a code conflict I need to resolve?

**Comment by sdhzwc** [comment]
@ruanwenjun That should do it.

**Review by ruanwenjun — APPROVED** [review]
Backend LGTM

**Review by SbloodyS — APPROVED** [review]
+1

**Comment by SbloodyS** [comment]
Please fix UT. @sdhzwc

**Pull request merged**